### PR TITLE
fix: Correctly handle pagination when other query parameters are set

### PR DESCRIPTION
**Pull request by UnseenWizzard**
The addition of an extra query parameter for getting all SLO configurations, lead to discovering that we do not correctly handle pagination if any other query params are set.

Also extend the previously added to tests to fully validate the behaviour of the method getting all configs from the API, encluding what parameters are set based on defined server returns and whether we expect an error or success.

### CODE REVIEW

File: pkg/rest/config_upload_test.go [hunk 1]
```diff
@@ -424,36 +424,151 @@ func Test_retryReturnContainsHttpErrorIfNotSuccess(t *testing.T) {
...
+
+				resp := tt.serverResponses[apiCalls]
+				if resp.statusCode != 200 {
+					http.Error(rw, resp.body, resp.statusCode)
+				} else {
+					_, _ = rw.Write([]byte(resp.body))
+				}
+
+				apiCalls++
+				assert.Check(t, apiCalls <= tt.expectedApiCalls, "expected at most %d API calls to happen, but encountered call %d", tt.expectedApiCalls, apiCalls)
```
Comment (UnseenWizzard) [inline pkg/rest/config_upload_test.go:559]: this might go a bit far into testing the exact behaviour/flow of execution - what do you think?

**Review by warber — COMMENTED** [review]
One nitpick that might improve readability a tiny bit:
In `getExistingValuesFromEndpoint` we want to work with the url as a dedicated type (`*url.URL`).
So we could do the parsing of the url as a first step here and pass through (and actually also handle the error):

``` go
func getExistingValuesFromEndpoint(client *http.Client, theApi api.Api, urlString string, apiToken string) (values []api.Value, err error) {
        // lets parse it already here and handle the error
	url, err := url.Parse(urlString)
	if err != nil {
		return nil, err
	}
        // --> lets pass the url as *url.URL (the function anyway needs it like that)
       url = addQueryParamsForNonStandardApis(theApi, url)
...

```

File: pkg/rest/config_upload.go [hunk 2]
```diff
@@ -369,19 +369,19 @@ func isMobileApp(api api.Api) bool {
 	return api.GetId() == "application-mobile"
 }
 
-func getExistingValuesFromEndpoint(client *http.Client, theApi api.Api, url string, apiToken string) (values []api.Value, err error) {
+func getExistingValuesFromEndpoint(client *http.Client, theApi api.Api, urlString string, apiToken string) (values []api.Value, err error) {
 
-	url = addQueryParamsForNonStandardApis(theApi, url)
+	u := addQueryParamsForNonStandardApis(theApi, urlString)
 
 	var existingValues []api.Value
```
[the remark this answers is not in the document]
Comment (UnseenWizzard) [reply pkg/rest/config_upload.go]: addressed in fixup :)

**Review by warber — APPROVED** [review]
lgtm

**Review by Laubi — APPROVED** [review]

**Pull request merged**